ruma-client-api/src/r0/capabilities/get_capabilities/iter.rs

```diff
@@ -0,0 +1,63 @@
+use std::{borrow::Cow, collections::btree_map};
+
+use serde_json::Value as JsonValue;
+
+use super::Capabilities;
+
+/// Reference to a capability
+#[derive(Debug)]
+pub struct CapabilityRef<'a> {
+    name: &'a str,
+    value: Option<&'a JsonValue>,
+    caps: &'a Capabilities,
+}
+
+impl<'a> CapabilityRef<'a> {
+    pub fn name(&self) -> &'a str {
+        self.name
+    }
+
+    pub fn value(&self) -> Cow<'a, JsonValue> {
+        match self.value {
+            // unknown capability from btreemap iterator
+            Some(val) => Cow::Borrowed(val),
+            // O(1) lookup of known capability
+            None => self.caps.get(self.name).unwrap(),
+        }
+    }
+}
+
+/// An iterator over capabilities
+#[derive(Debug)]
+pub struct CapabilitiesIter<'a> {
+    caps: &'a Capabilities,
+    pos: usize,
+    custom_caps_iterator: btree_map::Iter<'a, String, JsonValue>,
+}
+
+impl<'a> CapabilitiesIter<'a> {
+    pub fn new(caps: &'a Capabilities) -> Self {
```

Comment:
Assuming that this module will be made public, this function's visibility should probably be restricted.

```suggestion
    pub(in super) fn new(caps: &'a Capabilities) -> Self {
```